shared/lib/sharing/publicize/actions.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	FETCH_PUBLICIZE_CONNECTIONS,
+	RECEIVE_PUBLICIZE_CONNECTIONS,
+	FAIL_PUBLICIZE_CONNECTIONS_REQUEST
+} from 'lib/redux-store/action-types';
+
+/**
+ * Triggers a network request to fetch Publicize connections for the specified
+ * site ID.
+ *
+ * @param  {Number}   siteId Site ID
+ * @return {Function}        Action thunk
+ */
+export function fetchConnections( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: FETCH_PUBLICIZE_CONNECTIONS,
+			siteId
+		} );
+
+		return new Promise( ( resolve ) => {
```

Comment:
>@aduth: I haven't examined the PR closely enough, so might as well ask directly: is the promise returned here caught and handled anywhere? Or were you considering catching promises via middleware (e.g. `redux-promised-thunk`)?

The promise isn't entirely necessary and is not currently used. The thought behind returning a promise was that it would be easier to handle in the context of server-side rendering for determining the point at which all data needs have been satisfied (i.e. once the aggregated set of all promises from thunk'd actions have been settled). I may have been a bit premature here, since we've not decided on what this will end up looking like. Perhaps it's best that I remove the promise for now.